### ISSUE TITLE
Sending messages

### ISSUE THREAD

**Issue by jtpio** [vidartf/jupyterlab-kernelspy#11]
It would be great to be able to send custom messages to the kernel.

We could imagine having a new panel with a few options:

1. to select the channel (`shell`, `control`...)
2. select the message (`request_execute`, `debug_request` and so on)
3. have a json blob populated with the right schema (based on to step 1 and 2)
4. click on a "Send" button to send the message
5. use the existing `KernelSpyView` to inspect the messages

**Comment by vidartf** [comment]
> We could imagine having a new panel [...]:

I like the idea, but this could probably be a separate extension then? Or would there be some significant overlap with the current codebase?

**Comment by jtpio** [comment]
If it's in a different extension then is it reasonable to assume users would also expect being able to inspect the messages being sent (which is already handled by the kernel spy).

Maybe as a different plugin otherwise?

**Comment by vidartf** [comment]
Ah, yeah I guess there is the dist concerns while waiting for a way for jupyterlab extensions to require other extensions (there is a jupyterlab issue about that, but I cannot find it atm).

>Maybe as a different plugin otherwise?

I think that would make sense. Possibly the plugins could be split into a file each as well.

**Comment by jtpio** [comment]
Yes, the related JupyterLab issues are https://github.com/jupyterlab/jupyterlab/issues/7064 and https://github.com/jupyterlab/jupyterlab/issues/4245.
